table_stats.go

```diff
@@ -240,20 +240,23 @@ func (d *DB) scanReadStateTableStats(
 
 			// If the file is remote and not SharedForeign, we should check if its size
 			// matches. This is because checkConsistency skips over remote files.
-			// SharedForeign files are skipped as their sizes are allowed to have a
-			// mismatch; the size stored in the FileBacking is just the part of the
-			// file that is referenced by this Pebble instance, not the size of the
-			// whole object.
+			//
+			// SharedForeign and External files are skipped as their sizes are allowed
+			// to have a mismatch; the size stored in the FileBacking is just the part
+			// of the file that is referenced by this Pebble instance, not the size of
+			// the whole object.
 			objMeta, err := d.objProvider.Lookup(fileTypeTable, f.FileBacking.DiskFileNum)
 			if err != nil {
 				// Set `moreRemain` so we'll try again.
 				moreRemain = true
 				d.opts.EventListener.BackgroundError(err)
 				continue
 			}
-			if _, ok := sizesChecked[f.FileBacking.DiskFileNum]; !ok && objMeta.IsRemote() &&
-				!d.objProvider.IsSharedForeign(objMeta) {
 
+			shouldCheckSize := objMeta.IsRemote() &&
```

Comment:
nit: this could also just be `shouldCheckSize := objMeta.IsShared() && !d.objProvider.IsSharedForeign(objMeta)`, but maybe it's better to have a blacklist rather than a whitelist of things to check so to speak. Good with it either way.